core/transactor/tracker/tracker.go

```diff
@@ -66,13 +65,18 @@ func (t *Tracker) trackStatus(ctx context.Context, resp *Response) {
 			// If the context is done, it could be due to cancellation or other reasons.
 			return
 		case <-timer.C:
-			// Not found in mempool, wait for it to be mined or go stale.
+			// Not found in pending mempool, wait for it to be mined or go stale.
```

Comment:
note: only need to check if a tx is in pending. if a tx has been queued, we can just wait for the configured timeout to see if it moves back into pending or it will stay in queued and its gonna be stale.